--- a/src/com/ibm/zurich/idmx/showproof/Verifier.java.orig	2013-05-02 12:14:56.689453117 +0200
+++ b/src/com/ibm/zurich/idmx/showproof/Verifier.java	2013-05-02 12:18:05.989440173 +0200
@@ -71,7 +71,7 @@
    * @return Verifier's nonce.
    */
   public static BigInteger generateNonce(final SystemParameters sp) {
-    return Utils.computeRandomNumberSymmetric(sp.getL_m());
+    return Utils.computeRandomNumberSymmetric(sp.getL_Phi());
   }
 
   private static void getMHatBounds(final SystemParameters sp, final BigInteger bounds[]) {
